src/cms/views/utils/search_content_ajax.py

```diff
@@ -0,0 +1,103 @@
+import re
+import logging
+import json
+
+from django.contrib.auth.decorators import login_required
+from django.http import JsonResponse
+from django.views.decorators.http import require_POST
+
+from backend.settings import WEBAPP_URL
+from ...decorators import region_permission_required
+from ...models.regions.region import Region
+
+logger = logging.getLogger(__name__)
+
+# The maximum number of results returned by `search_content_ajax`
+MAX_RESULT_COUNT = 20
+
+
+@require_POST
+@login_required
+@region_permission_required
+# pylint: disable=unused-argument
+def search_content_ajax(request, region_slug, language_slug):
+    """Searches all pois, events and pages for the current region and returns all that
+    match the search query. Results which match the query in the title or slug get ranked
+    higher than results which only match through their text content.
+
+    :param request: The current request
+    :type request: ~django.http.HttpResponse
+    :param region_slug: region identifier
+    :type region_slug: str
+    :param language_slug: language slug
+    :type language_slug: str
+    :return: Json object containing all matching elements, of shape {title: str, url: str, type: str}
+    :rtype: ~django.http.JsonResponse
+    """
+
+    def format_object_translation(object_translation, typ):
+        """Formats the [poi/event/page]-translation as json"""
+        return {
+            "title": object_translation.title,
+            "url": f"{WEBAPP_URL}/{object_translation.permalink}",
+            "type": typ,
+        }
+
+    def find_objects(object_manager, query, language_slug, body_name):
+        """Filters all object translations from ``object_manager`` of this language
+        that match ``query`` and returns them and their ranking"""
+        for obj in object_manager.all():
+            if obj.archived:
+                continue
```

Comment:
I think it might be a bit inefficient to query all objects first, then check whether they're archived, then do a regex matching on the translation fields...
Maybe we can do the search on the SQL level with the Django QuerySet features (see [docu](https://docs.djangoproject.com/en/3.2/topics/db/search/))? I'm imagining something like:
```
PageTranslation.objects.filter(
    Q(page__region=region, language=language) & 
    (Q(slug__unaccent__icontains=query) | Q(title__unaccent__icontains=query))
).distinct("page")
```
But that's probably hard to get right and generalize for all content types (and we have the problem with the priorities, but if it's too complicated I'd rather remove the body-search). I will think about this and get back to you in a few days.

Reply:
Okay. I have removed the body search for now.